tests/test_notifications.py

```diff
@@ -0,0 +1,43 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2023 Graz University of Technology.
+#
+# Invenio-Users-Resources is free software; you can redistribute it and/or
+# modify it under the terms of the MIT License; see LICENSE file for more
+# details.
+
+"""Notification related tests."""
+
+from copy import deepcopy
+
+from invenio_notifications.models import Notification, Recipient
+
+from invenio_users_resources.notifications import UserPreferencesRecipientFilter
+from invenio_users_resources.records.api import UserAggregate
+
+
+def test_user_recipient_filter(user_pub):
+    """Test user recipient filter for notifications."""
+    preferences_filter = UserPreferencesRecipientFilter()
+
+    u = UserAggregate.from_user(user_pub.user).dumps()
+
+    user_notifications_enabled = deepcopy(u)
+    user_notifications_enabled["preferences"]["notifications"] = {"enabled": True}
```

Comment:
minor: could we maybe override the `ACCOUNTS_USER_PREFERENCES_SCHEMA` in conftest.py to integrate `NotificationPreferences` to test this?

Reply:
adapted conftest.py to use a preferences schema with `NotificationPreferences`.